frontend/src/cljs/imas_seamap/map/events.cljs

```diff
@@ -31,50 +31,61 @@
-                                             (map {:imagery     0
-                                                   :habitat     1
-                                                   :third-party 2})
-                                             (apply min))]]
-                     (let [info-format (get-layers-info-format active-layers)
-                           params {:REQUEST       "GetFeatureInfo"
-                                   :LAYERS        layers
-                                   :QUERY_LAYERS  layers
-                                   :WIDTH         (:x size)
-                                   :HEIGHT        (:y size)
-                                   :BBOX          (bounds->str bounds)
-                                   :FEATURE_COUNT 5
-                                   :STYLES        ""
-                                   :X             x
-                                   :Y             y
-                                   :TRANSPARENT   true
-                                   :CRS           "EPSG:4326"
-                                   :SRS           "EPSG:4326"
-                                   :FORMAT        "image/png"
-                                   :INFO_FORMAT   info-format
-                                   :SERVICE       "WMS"
-                                   :VERSION       "1.1.1"}]
-                       {:method          :get
-                        :uri             server-url
-                        :params          params
-                        :response-format (ajax/text-response-format)
-                        :on-success      [:map/got-featureinfo request-id priority point info-format]
-                        :on-failure      [:map/got-featureinfo-err request-id priority]}))
+      (cond->
        ;; Initialise marshalling-pen of data: how many in flight, and current best-priority response
-       :db         (assoc db :feature-query {:request-id        request-id
-                                             :response-remain   (count by-server)
-                                             :response-priority 99
-                                             :had-insecure?     had-insecure?
-                                             :candidate         nil}
-                          :feature       {:status   :feature-info/waiting
-                                          :location point})}
+       {:db (assoc db
+                   :feature-query {:request-id        request-id
+                                   :response-remain   (count by-server)
+                                   :response-priority 99
+                                   :had-insecure?     had-insecure?
+                                   :candidate         nil}
+                   :feature       {:status   :feature-info/waiting
+                                   :location point})}
+        
+        info-format
+        (assoc
+         :http-xhrio
+         (when info-format
```

Comment:
This looks redundant -- isn't the cond-> clause "info-format" already implicitly a "truthy" condition?

Reply:
In fact looking further, I think this would be better extracted as a function at this point -- you're using cond-> to implement if-then-else, and the whole lot is getting a bit hard to follow.

Reply:
For your first comment, yep, that's silly me who forgot to delete some code I made when trying out solutions. For your second comment, I'll work on that now.